### PR TITLE
Load statistics data from server (EXPOSUREAPP-4514)

### PR DESCRIPTION
**Pull request by d4rken**
This replaces the mock data with data loaded from the servers.

* Respect cache header from server
* Verify data signature
* Use local fallback cache
* Refresh on app foreground/background state
* Filter out unsupported data types
* Sort data by card order received from server
* Check card data for validity (are all key figures there?)

### TODO
- [x] Complete unit  tests

### Testing
* Only DEV and INT currently has statistics data
* Filter logcat for "Statistics"
* Check against TechSpec

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/statistics/StatsItem.kt [hunk 1]
```diff
@@ -1,30 +1,117 @@
...
 }
 
-sealed class StatsItem(val cardId: Int) {
+sealed class StatsItem(val cardType: Type) {
     abstract val updatedAt: Instant
-    abstract val keyFigures: List<KeyFigureCardOuterClass.KeyFigure>
+    abstract val keyFigures: List<KeyFigure>
+
+    enum class Type(val id: Int) {
+        INFECTION(1),
```
Comment (ralfgehrer) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/statistics/StatsItem.kt:22]: minor: might be a little odd to start with 1 instead of 0

Comment (ralfgehrer) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/statistics/StatsItem.kt:22]: Or is this referring to the `cardIdSequence` with values `[1,2,3,4]` of the tech spec?

Comment (d4rken) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/statistics/StatsItem.kt:22]: Yes. These are the `cardId`'s defined by the tech spec.

**Review by BMItr — APPROVED** [review]
lgtm 👌

**Review by ralfgehrer — APPROVED** [review]
LGTM

**Pull request merged**